### PR TITLE
refactor(qwen35): dedupe the TP decode arms and shard-loader checks

### PR DESCRIPTION
**Pull request by FeathBow**
## Description

Cleanup.

Invariant touched: none of the serving contracts. The row shard still uploads its slice of the borrowed payload without a host copy; the TP decode arms produce the same rows, tokens and logprobs from the same buffers; the pre-capture watchdog keeps disarm-on-success / armed-on-error.

- `pegainfer-core`: `tensor_2d_dims` / `check_row_range` replace three copies of the 2D shape check and two of the row bounds check; `load_tensor_1d_shard` (one-line wrapper, one caller) removed; both F32 loaders share `upload_f32`.
- `pegainfer-qwen35`: `sample_decode_rows` and `states_in_row_order` serve both the eager and graph decode arms; the "sampler may modify bufs.logits in place" comment removed (the sampler takes the arena by shared reference); watchdog `match` collapsed to `?`; `LinearStatePointerTables.batch_size` renamed `capacity`.
- Tests removed: `tp_cuda_graph_is_no_longer_rejected_before_model_load`, `tp_engine_cuda_graph_passes_preload_validation`, `tensor_parallel_cuda_graph_gate_defers_to_model_load` (assert the absence of a removed error string on a bogus path; any load error passes), `linear_conv1d_shard_segments_scale_by_kernel_dim` (asserts literals of a one-line map).
- Docs: "Rebase onto #870" section dropped (branch history; cited tests removed in the #1003 review), P2b acceptance no longer pinned to a pre-rebase SHA, heading collision with P2a "Step 3" fixed, dangling knowledge-eval link and P2c process notes removed, index row no longer quotes pre-rebase throughput numbers.

## Test Env

Single GPU (sm_89, x86_64), Qwen3.5-4B.

## Verification

Validation host: Single GPU (sm_89, x86_64), worktree at this branch head.

| Check | Result |
| --- | --- |
| `cargo fmt --all -- --check` | clean |
| `cargo clippy --release -p pegainfer-core --all-targets -- -D warnings` | clean |
| `cargo clippy --release -p pegainfer-qwen35 --features qwen35 --all-targets -- -D warnings` | clean |
| `cargo check --release -p pegainfer-qwen3 --lib` (weight_loader consumer) | clean |
| `cargo test --release -p pegainfer-core --lib weight_loader` | 5 passed |
| `cargo test --release -p pegainfer-qwen35 --features qwen35 --lib` | 98 passed, 7 ignored (GPU-gated) |
| `hf_golden_gate` short, Qwen3.5-4B, single GPU | 108 positions, mean 0.0256 / p99 0.1166 / max 0.1794 (sequential graph; batched 5-padded and 3-padded; slot-compaction) |
| `hf_golden_gate` long (4097 / 8192), Qwen3.5-4B | 18 positions, mean 0.0219 / p99 0.0691 / max 0.0774 |

Not run here: TP2 gates (`pega_logprobs_match_hf_golden_within_qwen35_tolerance_tp2{,_graph}`, `test_e2e_qwen35_scheduler_tp2{,_graph}`, `serving_tp2`)

### CODE REVIEW

**Comment by FeathBow** [comment]
@codex review.
